code/service/src/main/scala/com/namely/chiefofstate/StartNodeBehaviour.scala

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2020 Namely Inc.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+package com.namely.chiefofstate
+
+import akka.NotUsed
+import akka.actor.typed.{ActorSystem, Behavior}
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.sharding.typed.scaladsl.{ClusterSharding, Entity}
+import akka.cluster.typed.Cluster
+import akka.management.cluster.bootstrap.ClusterBootstrap
+import akka.management.scaladsl.AkkaManagement
+import akka.persistence.typed.PersistenceId
+import akka.util.Timeout
+import com.namely.chiefofstate.config.{CosConfig, ReadSideConfigReader}
+import com.namely.chiefofstate.plugin.PluginManager
+import com.namely.chiefofstate.telemetry._
+import com.namely.protobuf.chiefofstate.v1.readside.ReadSideHandlerServiceGrpc.ReadSideHandlerServiceBlockingStub
+import com.namely.protobuf.chiefofstate.v1.service.ChiefOfStateServiceGrpc.ChiefOfStateService
+import com.namely.protobuf.chiefofstate.v1.writeside.WriteSideHandlerServiceGrpc.WriteSideHandlerServiceBlockingStub
+import com.typesafe.config.Config
+import io.grpc._
+import io.grpc.netty.NettyServerBuilder
+import io.opentracing.contrib.grpc.{TracingClientInterceptor, TracingServerInterceptor}
+import io.opentracing.util.GlobalTracer
+import org.slf4j.{Logger, LoggerFactory}
+
+import java.net.InetSocketAddress
+import scala.concurrent.ExecutionContext
+import scala.sys.ShutdownHookThread
+import scala.util.{Failure, Success, Try}
+
+/**
+ * This helps setup the required engines needed to smoothly run the ChiefOfState sevice.
+ * The following engines are started on boot.
+ * <ul>
+ *   <li> the akka cluster system
+ *   <li> the akka cluster sharding engine
+ *   <li> the akka cluster management
+ *   <li> loads the various ChiefOfState plugins
+ *   <li> run the required schemas and migration needed
+ *   <li> the telemetry tools and the various gRPC interceptors
+ *   <li> the gRPC service
+ * </ul>
+ */
+object StartNodeBehaviour {
+  final val log: Logger = LoggerFactory.getLogger(getClass)
+
+  def apply(config: Config): Behavior[NotUsed] = {
+    Behaviors.setup { context =>
+      // get the  COS config
+      val cosConfig: CosConfig = CosConfig(config)
+
+      val cluster: Cluster = Cluster(context.system)
+      context.log.info(s"starting node with roles: ${cluster.selfMember.roles}")
+
+      // Start the akka cluster management tool
+      AkkaManagement(context.system).start()
+      ClusterBootstrap(context.system).start()
+
+      Try {
+        // create data stores and run migrations if necessary
+        if (cosConfig.createDataStores) {
+          log.info("kick-starting the journal and snapshot store creation")
+          SchemasUtil.createIfNotExists(config)
+        } else {
+          log.info("No need to create stores")
+        }
+      } match {
+        case Failure(exception) => throw exception
```

Comment:
If you're going to throw it anyways, no need to wrap above in a `Try`. This function will execute serially by default.

Reply:
It makes sense